### ISSUE TITLE
page-break-inside: avoid -> specific column widths of table are ignored

### ISSUE THREAD

**Issue by gmeister2** [dompdf/dompdf#2395]
In version 1.0.2 the specific column widths of a table are ignored (see example below), when `page-break-inside: avoid` is triggered (i.e. when the table would be split between 2 pages if `page-break-inside: avoid` was omitted). The result is that the columns are evenly divided.

In version 0.8.5 the specific column widths are retained.

PHP 7.4 is used here.

```
<div class="page-break-inside-avoid">
    <h2>title</h2>

    <table class="table">
        <tbody>
        <tr>
            <td class="col-2">text</td>
            <td class="col-2">text</td>
            <td class="col-4">text</td>
            <td class="col-4">text</td>
        </tr>
        <tr>
            <td>text</td>
            <td>text</td>
	    <td>text</td>
	    <td>text</td>
        </tr>
	<tr>
            <td>text</td>
            <td>text</td>
	    <td>text</td>
	    <td>text</td>
        </tr>
        </tbody>
    </table>
</div>
```

```
.page-break-inside-avoid {
    page-break-inside: avoid;
}

.table {
    display: table;
    table-layout: fixed;
    width: 100%;
    max-width: 100%;
    margin-bottom: 30px;
    border-collapse: separate;
    border-spacing: 2px;
}

.col-2 {
    width: 16.66%;
}

.col-4 {
    width: 33.33%;
}
```

**Comment by gmeister2** [comment]
Possibly related to:
https://github.com/dompdf/dompdf/issues/2353

**Comment by bsweeney** [comment]
Do you have a full example? This is more likely related to issue #1432, but that's assuming that the same was seen in 0.8.5, which you indicate was not the case.

**Comment by bsweeney** [comment]
I can confirm, however, that fixed table layout does work on the first page. So this may be related but perhaps the issue is a bit more complex than I initially thought.

**Comment by bsweeney** [comment]
OK, it does look like fixed table layout may not be sticky across pages, though I think the issue is a little more complex. A simple table works OK for me in initial testing.

**Comment by gmeister2** [comment]
Thanks for your answers.

I made an example, with which you can test the issue (version 1.0.2).

Table 2 is where the issue occurs. Table 1 only serves to trigger it, this with the help of the variable $count, for example:
- if $count = 25 -> specific widths of Table 2 are ok
- if $count = 35 -> specific widths of Table 2 are not ok (columns are evenly divided)
- if $count = 45 -> specific widths of Table 2 are ok

I also double checked with version 0.8.5, there all 3 above scenarios work ok.

```
public function generateHtml(): string
    {
        ob_start();?>
        <html>
        <head>
            <style>
                .page-break-inside-avoid {
                    page-break-inside: avoid;
                }

                .table {
                    display: table;
                    table-layout: fixed;
                    width: 100%;
                    max-width: 100%;
                    margin-bottom: 30px;
                    border-collapse: separate;
                    border-spacing: 2px;
                }

                .col-2 {
                    width: 16.66%;
                }

                .col-4 {
                    width: 33.33%;
                }

                .grey {
                    background-color: grey;
                }

                .orange {
                    background-color: orange;
                }

                .green {
                    background-color: green;
                }
            </style>
        </head>
        <body>

        <h2>Table 1</h2>

        <?php $count = 35; ?>

        <table class="table">
            <tbody>
            <?php for ($i = 0; $i < $count; $i++): ?>
                <tr>
                    <td class="col-4 grey">col 4</td>
                    <td class="col-4 grey">col 4</td>
                    <td class="col-4 grey">col 4</td>
                </tr>
            <?php endfor; ?>
            </tbody>
        </table>

        <div class="page-break-inside-avoid">
            <h2>Table 2</h2>

            <table class="table">
                <tbody>
                <tr>
                    <td class="col-2 orange">col 2</td>
                    <td class="col-2 orange">col 2</td>
                    <td class="col-4 green">col 4</td>
                    <td class="col-4 green">col 4</td>
                </tr>
                <tr>
                    <td class="orange">col 2</td>
                    <td class="orange">col 2</td>
                    <td class="green">col 4</td>
                    <td class="green">col 4</td>
                </tr>
                <tr>
                    <td class="orange">col 2</td>
                    <td class="orange">col 2</td>
                    <td class="green">col 4</td>
                    <td class="green">col 4</td>
                </tr>
                <tr>
                    <td class="orange">col 2</td>
                    <td class="orange">col 2</td>
                    <td class="green">col 4</td>
                    <td class="green">col 4</td>
                </tr>
                <tr>
                    <td class="orange">col 2</td>
                    <td class="orange">col 2</td>
                    <td class="green">col 4</td>
                    <td class="green">col 4</td>
                </tr>
                </tbody>
            </table>
        </div>

        </body>
        </html>
        <?php
        $html = ob_get_clean();

        return trim($html);
    }
```

**Comment by Mellthas** [comment]
Fixed by #2502.

**Comment by uncaught** [comment]
I've run into the same issue and can confirm that #2502 fixes it! Can this get merged/released?
